staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_tls.go

```diff
@@ -228,16 +233,101 @@ func (o *CreateSecretTLSOptions) createSecretTLS() (*corev1.Secret, error) {
 	if err != nil {
 		return nil, err
 	}
+	var tlsCertAuthority []byte
+	if len(o.CertAuthority) != 0 {
+		tlsCertAuthority, err = readFile(o.CertAuthority)
+		if err != nil {
+			return nil, err
+		}
+	}
 	if _, err := tls.X509KeyPair(tlsCert, tlsKey); err != nil {
 		return nil, err
 	}
+
+	// validate that cert is a valid certificate
+	rest := tlsCert
+	var _tlsCert *x509.Certificate
+	var block *pem.Block
+	for {
+		block, rest = pem.Decode(rest)
+		if block == nil || block.Type != "CERTIFICATE" {
+			return nil, fmt.Errorf("failed to decode PEM block containing certificate.")
+		}
+		_tlsCert, err = x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse PEM block in certificate: %s", err)
+		}
+		if len(rest) == 0 {
+			break
+		}
+	}
+
+	// validate certAuthority is a valid certificate
+	rest = tlsKey
+	for {
+		block, rest = pem.Decode(rest)
+		if block == nil {
+			return nil, fmt.Errorf("failed to decode PEM block containing private key.")
+		}
+		if block.Type == "RSA PRIVATE KEY" {
+			_, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse PEM block in RSA private key, %s", err)
+			}
+		} else if block.Type == "EC PRIVATE KEY" {
+			_, err := x509.ParseECPrivateKey(block.Bytes)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse PEM block in EC private key: %s", err)
+			}
+		} else {
+			return nil, fmt.Errorf("unknown PEM block in private key: %s.", block.Type)
+		}
+		if len(rest) == 0 {
+			break
+		}
+	}
+
+	if len(tlsCertAuthority) > 0 {
+		// validate certAuthority is a valid certificate
+		rest = tlsCertAuthority
+		for {
+			block, rest = pem.Decode(rest)
+			if block == nil || block.Type != "CERTIFICATE" {
+				return nil, fmt.Errorf("failed to decode PEM block containing certificate authority.")
+			}
+			_, err := x509.ParseCertificate(block.Bytes)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse PEM block in certificate authority: %s.", err)
+			}
+			if len(rest) == 0 {
+				break
+			}
+		}
+
+		// validate that the certificate is valid with certAuthority
+		roots := x509.NewCertPool()
+		if !roots.AppendCertsFromPEM([]byte(tlsCertAuthority)) {
+			return nil, fmt.Errorf("could not use certificate authority for validating.")
+		}
+		// validate only valid root and (impliticly) generation time
+		opts := x509.VerifyOptions{
+			Roots: roots,
+		}
+
+		if _, err := _tlsCert.Verify(opts); err != nil {
+			return nil, fmt.Errorf("failed to verify certificate: %s", err.Error())
+		}
+	}
+
 	// TODO: Add more validation.
-	// 1. If the certificate contains intermediates, it is a valid chain.
-	// 2. Format etc.
+	// E.g., if the certificate contains intermediates, it is a valid chain.
 
 	secretTLS := newSecretObj(o.Name, namespace, corev1.SecretTypeTLS)
 	secretTLS.Data[corev1.TLSCertKey] = []byte(tlsCert)
 	secretTLS.Data[corev1.TLSPrivateKeyKey] = []byte(tlsKey)
+	if len(o.CertAuthority) != 0 {
```

Comment:
nit here: you can use len(tlsCertAuthority) instead of the configuration.

Reply:
Done.  Thanks.